bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/Tr064RootHandler.java

```diff
@@ -0,0 +1,391 @@
+
+    Tr064RootHandler(Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+        soapConnector = new SOAPConnector(httpClient, endpointBaseURL);
+    }
+
+    @Override
+    @SuppressWarnings("null")
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        Tr064ChannelConfig channelConfig = channels.get(channelUID);
+        if (channelConfig == null) {
+            logger.trace("Channel {} not supported.", channelUID);
+            return;
+        }
+
+        if (command instanceof RefreshType) {
+            State state = stateCache.putIfAbsentAndGet(channelUID,
+                    () -> soapConnector.getChannelStateFromDevice(channelConfig, channels, stateCache));
+            if (state != null) {
+                updateState(channelUID, state);
+            }
+            return;
+        }
+
+        if (channelConfig.getChannelTypeDescription().getSetAction() == null) {
+            logger.debug("Discarding command {} to {}, read-only channel", command, channelUID);
+            return;
+        }
+        scheduler.execute(() -> soapConnector.sendChannelCommandToDevice(channelConfig, command));
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(Tr064RootConfiguration.class);
+        if (!config.isValid()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "At least one mandatory configuration field is empty");
+            return;
+        }
+
+        endpointBaseURL = "http://" + config.host + ":49000";
+        updateStatus(ThingStatus.UNKNOWN);
+
+        connectFuture = scheduler.scheduleWithFixedDelay(this::internalInitialize, 0, RETRY_INTERVAL, TimeUnit.SECONDS);
+    }
+
+    /**
+     * internal thing initializer (sets SCPDUtil and connects to remote device)
+     */
+    private void internalInitialize() {
+        try {
+            scpdUtil = new SCPDUtil(httpClient, endpointBaseURL);
+        } catch (SCPDException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "could not get device definitions from " + config.host);
+            return;
+        }
+
+        if (establishSecureConnectionAndUpdateProperties()) {
+            final ScheduledFuture<?> connectFuture = this.connectFuture;
+            if (connectFuture != null) {
+                connectFuture.cancel(false);
+                this.connectFuture = null;
+            }
+
+            // connection successful, check channels
+            ThingBuilder thingBuilder = editThing();
+            thingBuilder.withoutChannels(thing.getChannels());
+            final SCPDUtil scpdUtil = this.scpdUtil;
+            if (scpdUtil != null) {
+                Util.checkAvailableChannels(thing, thingBuilder, scpdUtil, "", deviceType, channels);
+                updateThing(thingBuilder.build());
```

Comment:
The problem is that `Channel` does not implement `equals`, so it's difficult to check if the channels are really the same. Checking the `channelUID` is not enough, since the configuration could be different.